src/test/components/FilterNavigatorBar.test.js

```diff
@@ -0,0 +1,93 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import * as React from 'react';
+import { render } from 'react-testing-library';
+import { Provider } from 'react-redux';
+
+import ProfileFilterNavigator from '../../components/app/ProfileFilterNavigator';
+import * as ProfileView from '../../actions/profile-view';
+import { storeWithProfile } from '../fixtures/stores';
+import { getProfileFromTextSamples } from '../fixtures/profiles/processed-profile';
+
+describe('app/ProfileFilterNavigator', () => {
+  const browsingContextID = 123123;
+  function setup() {
+    const { profile } = getProfileFromTextSamples(`
+      A  A  A
+      B  B  B
+      C  C  H
+      D  F  I
+      E  G
+      `);
+    // Add page for active tab.
+    profile.pages = [
+      {
+        browsingContextID: browsingContextID,
+        innerWindowID: 1,
+        url: 'https://developer.mozilla.org/en-US/',
+        embedderInnerWindowID: 0,
+      },
+    ];
+    profile.meta.configuration = {
+      threads: [],
+      features: [],
+      capacity: 1000000,
+      activeBrowsingContextID: browsingContextID,
+    };
+
+    // Change the root range for testing.
+    const samples = profile.threads[0].samples;
+    samples.time[samples.length - 1] = 50;
+
+    const store = storeWithProfile(profile);
+    const renderResult = render(
+      <Provider store={store}>
+        <ProfileFilterNavigator />
+      </Provider>
+    );
+
+    return {
+      ...store,
+      ...renderResult,
+    };
+  }
+
+  it('renders ProfileFilterNavigator properly', () => {
+    const { container, dispatch } = setup();
+    // Just root range
+    expect(container.firstChild).toMatchSnapshot();
+
+    // With committed range
+    dispatch(ProfileView.commitRange(0, 40));
+    expect(container.firstChild).toMatchSnapshot();
+
+    // With preview selection
+    dispatch(
+      ProfileView.updatePreviewSelection({
+        hasSelection: true,
+        isModifying: false,
+        selectionStart: 10,
+        selectionEnd: 20,
+      })
+    );
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  it('displays the "Full Range" text as its first element', () => {
+    const { queryByText } = setup();
+    expect(queryByText('Full Range')).toBeTruthy();
+  });
+
+  it('displays the site hostname as its first element', () => {
+    const { dispatch, queryByText, container } = setup();
+    dispatch(ProfileView.changeShowTabOnly(browsingContextID));
+    // Checking the snapshot just in case, but we also have more specific checks below.
+    expect(container.firstChild).toMatchSnapshot();
```

Comment:
Also added a snapshot test here, but we also do specific checks later too. We can remove this shopshot if you want.

Reply:
I'd have the snapshot as a separate `it` block, so it fails separately from the more targeted tests.